File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/domain/did/SsdidRegistryResolver.kt
package my.ssdid.sdk.domain.did

import my.ssdid.sdk.domain.did.DidResolver
import my.ssdid.sdk.domain.model.DidDocument
import my.ssdid.sdk.domain.transport.RegistryApi

class SsdidRegistryResolver(private val registryApi: RegistryApi) : DidResolver {
    override suspend fun resolve(did: String): Result<DidDocument> = runCatching {
        registryApi.resolveDid(did)
    }
}

File: sdk/ios/SsdidCore/Sources/SsdidCore/Domain/Did/SsdidRegistryResolver.swift
import Foundation

public final class SsdidRegistryResolver: DidResolver {
    private let registryApi: RegistryApi

    public     init(registryApi: RegistryApi) {
        self.registryApi = registryApi
    }

    public     func resolve(did: String) async throws -> DidDocument {
        return try await registryApi.resolveDid(did: did)
    }
}
